File: solutions/patterns/star-patterns/12-hollow-inverted-pyramid/solution.cpp
#include <iostream>
using namespace std;

int main()
{
    int n;
    cout << "Enter the N value: ";
    cin >> n;

    cout << "\nPrinting the inverted pattern...\n\n";

    for (int row = n; row >= 1; row--)
    {
        // Print leading spaces
        for (int col = 1; col <= n - row; col++)
        {
            cout << " ";
        }

        if (row == 1)
        {
            // Last row (tip of the inverted pyramid)
            cout << "*";
        }
        else if (row == n)
        {
            // First row (widest row)
            for (int star = 1; star <= n; star++)
            {
                cout << "* ";
            }
        }
        else
        {
            // Middle rows
            cout << "*"; // Left star

            // Inner spaces
            for (int space = 0; space < (2 * row - 3); space++)
            {
                cout << " ";
            }

            cout << "*"; // Right star
        }

        cout << endl;
    }

    return 0;
}

File: solutions/patterns/star-patterns/12-hollow-inverted-pyramid/solution.js
let n = 5; // height of pyramid

for (let i = 0; i < n; i++) {
    let row = "";

    // spaces
    for (let j = 0; j <= i; j++) {
        row += " ";
    }

    // stars
    for (let k = 0; k < 2 * i + 1; k++) {
        if (k == 0 || k == 2 * i || (i == n - 1 && k % 2 == 0)) {
            row += "*";
        } else {
            row += " ";
        }
    }

    // print the row
    console.log(row);
}

//By The Architect 🙎‍♂️
